src/components/Popup.js

```diff
@@ -4,34 +4,38 @@ export default class Popup {
 
     this._popupElement = document.querySelector(popupSelector);
     this._closePopupButton = this._popupElement.querySelector(".modal__close");
-
-    // Define the event handlers separately
-    this._handleOverlayClick = (e) => this._handleOverlay(e);
-    this._handleDocumentKeydown = (e) => this._handleEscClose(e);
   }
 
-  _handleEscClose(e) {
-    if (e.key === "Escape") {
+  _handleDocumentKeydown(evt) {
+    if (evt.key === "Escape") {
       this.close();
     }
   }
 
-  _handleOverlay(evt) {
+  _handleOverlayClick(evt) {
     if (evt.target.id === this._popupElement.id) {
       this.close();
     }
   }
 
   open() {
     this._popupElement.classList.add("modal_opened");
-    document.addEventListener("click", this._handleOverlayClick, true);
-    document.addEventListener("keydown", this._handleDocumentKeydown, false);
+    document.addEventListener("click", (e) => {
+      this._handleOverlayClick(e);
+    });
+    document.addEventListener("keydown", (e) => {
+      this._handleDocumentKeydown(e);
```

Comment:
You shouldn't have changed the code of `open` and `close` because now it's incorrect. You can't remove listeners in such a way if you add them like this